NRLMMD-GEOIPS/geoips
# v1.6.0 geoips - VERSION RELEASE - data_fusion support, simplify installation, unique invest dirs

# Related Issues
required to close NRLMMD-GEOIPS/geoips#51

NOTE: This started as v1.5.4 release, but changed to v1.6.0 due to significant data_fusion support updates.

# Reviewer Instructions
* Please confirm most recent completed pull request is labeled "DEVELOPMENTAL UPDATE" or "VERSION RELEASE":
    * https://github.com/NRLMMD-GEOIPS/geoips/pulls?q=is%3Apr+is%3Aclosed
* Review "Files changed" tab
* Confirm appropriate testing was completed for these updates.

# Summary
Changes for version 1.6.0 update on repo geoips.
* updates for data_fusion support
* simplify installation and testing
* allow creating unique invest subdirectories (with storm start datetime, to disambiguate between repeated invest numbers)

See NRLMMD-GEOIPS/geoips#51 for additional repositories included in this update.

# Testing Instructions

Install GEOIPS and clone v1.6.0-release (or v1.5.4-release) branch for:
* geoips
* test_data_amsr2


Tests should return 0:
```
$GEOIPS_PACKAGES_DIR/geoips/setup.sh setup_abi_test_data  # To get other ABI channels

$GEOIPS_PACKAGES_DIR/geoips/tests/test_base_install.sh

$GEOIPS_PACKAGES_DIR/geoips/tests/scripts/amsr2.config_based_overlay_output_low_memory.sh
$GEOIPS_PACKAGES_DIR/geoips/tests/scripts/amsr2.config_based_overlay_output.sh
$GEOIPS_PACKAGES_DIR/geoips/tests/scripts/amsr2_ocean.tc.windspeed.imagery_clean.sh
$GEOIPS_PACKAGES_DIR/geoips/tests/scripts/amsr2.tc.89H-Physical.imagery_annotated.sh

$GEOIPS_PACKAGES_DIR/geoips/tests/scripts/abi.config_based_output_low_memory.sh
$GEOIPS_PACKAGES_DIR/geoips/tests/scripts/abi.config_based_output.sh
$GEOIPS_PACKAGES_DIR/geoips/tests/scripts/abi.static.Infrared.imagery_annotated.sh
$GEOIPS_PACKAGES_DIR/geoips/tests/scripts/abi.static.Visible.imagery_annotated.sh

```

# Output


```
Fri Dec 16 13:26:27 PST 2022  Running post, final results in $HOME/geoproc/test_data/logs/20221216/20221216.212221_geoips_base/test_all_geoips_base.log

Package: geoips_base
Total run time: 246 seconds
Number data types run: 3
Number data types failed: 0

full return: 0
```
```
16_155629 config_based1052   INFO: The following products were produced from procflow config_based.py
16_155629 config_based1058   INFO: SUCCESSFUL COMPARISON DIR: $HOME/geoproc/geoips_packages/geoips/tests/outputs/amsr2.tc_overlay.37pct.imagery_annotated_over_Infrared-Gray

16_155629 config_based1061   INFO:     CONFIGSUCCESS $HOME/geoproc/geoips_outdirs/preprocessed/tcwww/tc2020/IO/IO012020/png/37pct/gcom-w1/20200518_073601_IO012020_amsr2_gcom-w1_37pct_140kts_95p89_res1p0-cr100-bgInfrared-Gray.png
16_155629 config_based1061   INFO:     CONFIGSUCCESS $HOME/geoproc/geoips_outdirs/preprocessed/tcwww/tc2020/IO/IO012020/metadata/sector_information/20200518/20200518_073601_IO012020_amsr2_gcom-w1_37pct_140kts_95p89_res1p0-cr100-bgInfrared-Gray.png.yaml
16_155629 config_based1064   INFO:

16_155629 config_based1058   INFO: SUCCESSFUL COMPARISON DIR: $HOME/geoproc/geoips_packages/geoips/tests/outputs/amsr2.tc_overlay.89pct.imagery_annotated_over_Infrared-Gray

16_155629 config_based1061   INFO:     CONFIGSUCCESS $HOME/geoproc/geoips_outdirs/preprocessed/tcwww/tc2020/IO/IO012020/png/89pct/gcom-w1/20200518_073601_IO012020_amsr2_gcom-w1_89pct_140kts_98p32_res1p0-cr100-bgInfrared-Gray.png
16_155629 config_based1061   INFO:     CONFIGSUCCESS $HOME/geoproc/geoips_outdirs/preprocessed/tcwww/tc2020/IO/IO012020/metadata/sector_information/20200518/20200518_073601_IO012020_amsr2_gcom-w1_89pct_140kts_98p32_res1p0-cr100-bgInfrared-Gray.png.yaml
16_155629 config_based1064   INFO:

16_155629 config_based1058   INFO: SUCCESSFUL COMPARISON DIR: $HOME/geoproc/geoips_packages/geoips/tests/outputs/amsr2.global_overlay.89pct.imagery_annotated_over_Infrared-Gray

16_155629 config_based1061   INFO:     CONFIGSUCCESS $HOME/geoproc/geoips_outdirs/preprocessed/annotated_imagery/Global-x-x/x-x-x/89pct/amsr2/20200518.062048.gcom-w1.amsr2.89pct.global.13p55.star.20p0.png
16_155629 config_based1064   INFO:

16_155629 config_based1058   INFO: SUCCESSFUL COMPARISON DIR: $HOME/geoproc/geoips_packages/geoips/tests/outputs/amsr2.global_overlay.37pct.imagery_annotated_over_Infrared-Gray

16_155629 config_based1061   INFO:     CONFIGSUCCESS $HOME/geoproc/geoips_outdirs/preprocessed/annotated_imagery/Global-x-x/x-x-x/37pct/amsr2/20200518.062048.gcom-w1.amsr2.37pct.global.10p06.star.20p0.png

16_155629 config_based1087   INFO: READER_NAME: amsr2_netcdf
16_155629 config_based1088   INFO: NUM_PRODUCTS: 6
16_155629 config_based1091   INFO: NUM_DELETED_PRODUCTS: 0
16_155629 config_based1092   INFO: NUM_SUCCESSFUL_COMPARISON_DIRS: 4
16_155629 config_based1093   INFO: NUM_FAILED_COMPARISON_DIRS: 0
16_155629        utils 107   INFO: Total Time 0:01:26.229414: 4 Num jobs: GeoIPS 2
16_155629        utils 115   INFO:     SUCCESS Process Time GeoIPS 2: tc2020io01amphan    : 0:00:36.809009
16_155629        utils 115   INFO:     SUCCESS Process Time GeoIPS 2: global              : 0:00:39.444228
16_155629 run_procflow  46   INFO: Completed geoips PROCFLOW config_based processing, done!
16_155629 run_procflow  47   INFO: Starting time: 2022-12-16 23:54:54.279533
16_155629 run_procflow  48   INFO: Ending time: 2022-12-16 23:56:29.871407
16_155629 run_procflow  49   INFO: Total time: 0:01:35.591958
16_155629 run_procflow  61   INFO: Return value: 0

```

# Post Merge Steps
Once this PR has been merged, v1.6.0 will be tagged/released on the geoips main branch.
